### PR TITLE
34781 - Added submissionData to the context

### PR DESCRIPTION
**Pull request by marijnritense**

### CODE REVIEW

File: form-flow-valtimo/build.gradle [hunk 1]
```diff
@@ -44,6 +44,8 @@ dependencies {
     implementation "org.jetbrains.kotlin:kotlin-stdlib-jdk8"
     implementation "org.jetbrains.kotlin:kotlin-reflect"
 
+    implementation "org.json:json:20201115"
```
[the remark this answers is not in the document]
Comment (marijnritense) [reply form-flow-valtimo/build.gradle:47]: This is the latest version we're using in Valtimo. Not a huge fan of it either to be honest. What's the alternative? Just change it to the latest version and accept we're now using 4 different versions?

File: form-flow/src/main/kotlin/com/ritense/formflow/domain/instance/FormFlowInstance.kt [hunk 2]
```diff
@@ -95,6 +95,47 @@ class FormFlowInstance(
...
+        }
+
+        val mergedSubmissionData = formFlowStepInstancesSubmissionData.first()
+
+        if (formFlowStepInstancesSubmissionData.size == 1) {
+            return mergedSubmissionData.toString()
+        }
+
+        formFlowStepInstancesSubmissionData.subList(1, formFlowStepInstancesSubmissionData.size).forEach {
+            mergeSubmissionData(it, mergedSubmissionData)
```
[the remark this answers is not in the document]
Comment (marijnritense) [reply form-flow/src/main/kotlin/com/ritense/formflow/domain/instance/FormFlowInstance.kt:112]: We can open that discussion again if you'd like, but we did decide on this approach when we refined the story. Keys do not have to be unique, that just means you're overwriting an existing value.

File: form-flow-valtimo/build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (theo-ritense) [reply form-flow-valtimo/build.gradle:47]: Ideally: upgrade all versions of it to the latest. By having 4 different versions in our dependency tree, we could have random issues with different module combinations where one forces a different version than the other.

@ivo-ritense ?

File: form-flow/src/main/kotlin/com/ritense/formflow/domain/instance/FormFlowInstance.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (theo-ritense) [reply form-flow/src/main/kotlin/com/ritense/formflow/domain/instance/FormFlowInstance.kt:112]: Ok. Just checking if it was intentional.

File: form-flow-valtimo/build.gradle (hunk 1 above)
[the remark this answers is not in the document]
Comment (marijnritense) [reply form-flow-valtimo/build.gradle:47]: So small thing to keep in mind here: This (potentially) changes the scope of this story quite a bit. I do agree we should try and keep versions up to date, and I can definitely change this to a newer version to see if it still works. The other versions in our dependency tree come from other libraries. Maybe we should write it down as a separate story?

[the remark this answers is not in the document]
Comment (theo-ritense) [reply form-flow-valtimo/build.gradle:47]: Let's create a story on upgrading outdated dependencies.

**Review by theo-ritense — APPROVED** [review]

**Pull request merged**